From 819e637282112a5fa8ea232f7cc3f3b51dfe4317 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 12 Apr 2022 13:58:26 +0200
Subject: [PATCH 205/954] refactor(st-bsec): remove bsec_read_scratch

Remove the unused function bsec_read_scratch().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9fd81de35c4ffa87fc755805d5e079005b599322 refactor(st-bsec): remove bsec_read_scratch]
Change-Id: I9c9c96dd58d6e1d33e9b5f0f4d206df3a4e12223
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/250947
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/bsec/bsec2.c   | 8 --------
 include/drivers/st/bsec.h | 1 -
 2 files changed, 9 deletions(-)

diff --git a/drivers/st/bsec/bsec2.c b/drivers/st/bsec/bsec2.c
index 193747d0bc..7706d3b89a 100644
--- a/drivers/st/bsec/bsec2.c
+++ b/drivers/st/bsec/bsec2.c
@@ -532,14 +532,6 @@ void bsec_write_scratch(uint32_t val)
 #endif
 }
 
-/*
- * bsec_read_scratch: return scratch register value.
- */
-uint32_t bsec_read_scratch(void)
-{
-	return mmio_read_32(BSEC_BASE + BSEC_SCRATCH_OFF);
-}
-
 /*
  * bsec_get_status: return status register value.
  */
diff --git a/include/drivers/st/bsec.h b/include/drivers/st/bsec.h
index fc9d3d6ac1..8e072f756c 100644
--- a/include/drivers/st/bsec.h
+++ b/include/drivers/st/bsec.h
@@ -52,7 +52,6 @@ uint32_t bsec_program_otp(uint32_t val, uint32_t otp);
 uint32_t bsec_read_debug_conf(void);
 
 void bsec_write_scratch(uint32_t val);
-uint32_t bsec_read_scratch(void);
 
 /* Sticky lock support */
 uint32_t bsec_set_sr_lock(uint32_t otp);
-- 
2.53.0

